packages/google_maps_flutter/google_maps_flutter/test/polygon_updates_test.dart

```diff
@@ -122,14 +122,15 @@ void main() {
     expect(platformGoogleMap.polygonsToAdd.isEmpty, true);
   });
 
-  testWidgets("Mutate a polygon", (WidgetTester tester) async {
+  testWidgets('Mutate a polygon', (WidgetTester tester) async {
+    final List<LatLng> _points = <LatLng>[const LatLng(0.0, 0.0)];
     final Polygon p1 = Polygon(
-      polygonId: PolygonId("polygon_1"),
-      points: <LatLng>[const LatLng(0.0, 0.0)],
+      polygonId: const PolygonId('polygon_1'),
+      points: _points,
     );
     await tester.pumpWidget(_mapWithPolygons(<Polygon>{p1}));
 
-    p1.points.add(const LatLng(1.0, 1.0));
+    _points.add(const LatLng(1.0, 1.0));
```

Comment:
What was this set of changes fixing? The code seems less clear now, so I'm wondering if we can fix it differently.

Reply:
I meant to add `final List<LatLng> _points = <LatLng>[const LatLng(0.0, 0.0)];` to `p1`, to make _points list modifiable 
Otherwise, it throws `Unsupported operation: Cannot add to an unmodifiable list`


During local testing, I modified this line and forget to revert this line after making the p1 final above, thanks 

```diff
+   final List<LatLng> _points = <LatLng>[const LatLng(0.0, 0.0)];
     final Polygon p1 = Polygon(
-    polylineId: PolylineId("polyline_1"),
+   polygonId: const PolygonId('polygon_1'),
-    points: <LatLng>[const LatLng(0.0, 0.0)],
+   points: _points,
    );
    await tester.pumpWidget(_mapWithPolygons(<Polygon>{p1}));

    p1.points.add(const LatLng(1.0, 1.0));
```

Reply:
I see, the reason this code looks strange is because it is; looking at the `blame` this is testing that we notice if someone modifies what is supposed to be an immutable object.